compiler/circle-tensordump/src/Dump.cpp

```diff
@@ -0,0 +1,345 @@
+
+    auto tensors = reader.tensors();
+    for (const auto &tensor : *tensors)
+    {
+      os << std::string(50, '-') << std::endl;
+      os << "[" << tensor->name()->str() << "]" << std::endl;
+      auto buff_idx = tensor->buffer();
+      auto buff_data_ptr = reader.buffers()->Get(buff_idx)->data();
+      auto quant_param = tensor->quantization();
+      std::string print_format = (!buff_data_ptr && !quant_param) ? "└──" : "├──";
+
+      // shape
+      auto shape = tensor->shape();
+      os << " " + print_format + " shape : (";
+      ::print_comma_sepearted(os, shape);
+      os << ")" << std::endl;
+
+      // quantization paramters
+      if (quant_param)
+      {
+        std::string print_format1 = buff_data_ptr ? "├──" : "└──";
+        std::string print_format2 = buff_data_ptr ? "│" : " ";
+        os << " " + print_format1 + " quantization" << std::endl;
+        auto min = quant_param->min();
+        auto max = quant_param->max();
+        auto scale = quant_param->scale();
+        auto zero_point = quant_param->zero_point();
+
+        os << " " + print_format2 + "   ├── min        : ";
+        ::print_comma_sepearted(os, min);
+        os << std::endl;
+        os << " " + print_format2 + "   ├── max        : ";
+        ::print_comma_sepearted(os, max);
+        os << std::endl;
+        os << " " + print_format2 + "   ├── scale      : ";
+        ::print_comma_sepearted(os, scale);
+        os << std::endl;
+        os << " " + print_format2 + "   └── zero_point : ";
+        ::print_comma_sepearted(os, zero_point);
+        os << std::endl;
+      }
+
+      // buffer
+      print_buffer(os, buff_idx, buff_data_ptr, tensor->type());
+      os << std::endl;
+    }
+  }
+}
+
+} // namespace circletensordump
+
+namespace
+{
+
+// HDF5 forbids the inclusion of '/' in the name
+std::string mangle(const std::string &name)
+{
+  std::string ret{name};
+  std::replace(ret.begin(), ret.end(), '/', '_');
+  return ret;
+}
+
+H5::PredType hdf5_dtype_cast(const circle::TensorType &circle_type)
+{
+  switch (circle_type)
+  {
+    case circle::TensorType_UINT8:
+    {
+      return H5::PredType::NATIVE_INT8;
+    }
+    case circle::TensorType_INT32:
+    {
+      return H5::PredType::NATIVE_INT32;
+    }
+    case circle::TensorType_INT64:
+    {
+      return H5::PredType::NATIVE_INT64;
+    }
+    case circle::TensorType_FLOAT32:
+    {
+      ;
+      return H5::PredType::NATIVE_FLOAT;
+    }
+    default:
+      throw std::runtime_error("NYI tensor type : " + std::to_string(circle_type));
+  }
+}
+
+template <typename T>
+void write_data_to_hdf5(H5::H5File &file, const flatbuffers::Vector<T> *vec,
+                        const H5::PredType &type, std::string &group_name, std::string dataset_name)
+{
+  if (vec == nullptr)
+    return;
+  const uint32_t rank = vec->size();
+
+  // set dataset dimension
+  std::vector<hsize_t> dims(rank);
+  for (uint32_t d = 0; d < rank; d++)
+  {
+    dims[d] = vec->Get(d);
+  }
```

Comment:
Rank of vector seems always 1. `vec->size()` is the dimension of the 1D vector.

The code needs to be changed like below.

```
  std::vector<hsize_t> dims(1);
  dims[0] = vec->size();
  auto dataspace = std::make_unique<H5::DataSpace>(1, dims.data())
```

Reply:
It would be better to change the function name as well (something like `write_vector_to_hdf5` ?).

Reply:
Oh, it must have slipped my mind;ㅅ; I'll make it more general:)